client/src/components/SelectionDialog/HistoryDatasetPicker.vue

```diff
@@ -0,0 +1,325 @@
+
+        items.value = (data as HistorySummary[]).map(historyEntryToRecord);
+
+        formatRows();
+
+        return items.value;
+    } catch (error) {
+        errorMessage.value = errorMessageAsString(error);
+        return [];
+    } finally {
+        loading.value = false;
+    }
+}
+
+async function datasetsProvider(ctx: ItemsProviderContext, selectedHistory: HistoryRecord) {
+    loading.value = true;
+
+    try {
+        const limit = ctx.perPage;
+        const offset = (ctx.currentPage - 1) * ctx.perPage;
+        const query = ctx.filter || "";
+        const querySortBy = ctx.sortBy === "time" ? "update_time" : "name";
+        const sortPrefix = ctx.sortDesc ? "-dsc" : "-asc";
+
+        const { response, data, error } = await GalaxyApi().GET("/api/datasets", {
+            params: {
+                query: {
+                    history_id: selectedHistory.id,
+                    q: ["name-contains"],
+                    qv: [query],
+                    order: `${querySortBy}${sortPrefix}`,
+                    offset: offset,
+                    limit: limit,
+                },
+            },
+        });
+
+        if (error) {
+            errorMessage.value = errorMessageAsString(error);
+            return [];
+        }
+
+        totalItems.value = parseInt(response.headers.get("total_matches") ?? "0");
+
+        items.value = (data as HDASummary[]).map(datasetEntryToRecord);
+
+        formatRows();
+
+        return items.value;
+    } catch (error) {
+        errorMessage.value = errorMessageAsString(error);
+
+        return [];
+    } finally {
+        loading.value = false;
+    }
+}
+
+function onHistoryClick(item: SelectionItem) {
+    if (!item.isLeaf) {
+        selectionDialog.value?.resetFilter();
+        selectionDialog.value?.resetPagination();
+        datasetsVisible.value = true;
+        itemsProvider.value = (ctx: ItemsProviderContext) => datasetsProvider(ctx, item as HistoryRecord);
+    }
+}
+
+async function onDatasetClick(item: SelectionItem) {
+    if (item.isLeaf) {
+        if (selected.value.findIndex((i) => i.id === item.id) !== -1) {
+            selected.value = selected.value.filter((i) => i.id !== item.id);
+        } else {
+            selected.value.push(item);
+        }
+
+        formatRows();
+    }
+}
+
+function selectAll() {
+    if (allSelected.value) {
+        selected.value = [];
+    } else {
+        for (const item of items.value) {
+            if (selected.value.findIndex((i) => i.id === item.id) === -1) {
+                selected.value.push(item);
+            }
+        }
+    }
+
+    formatRows();
+}
+
+async function onUndo() {
+    selectionDialog.value?.resetFilter();
+    selectionDialog.value?.resetPagination();
+    selected.value = [];
+    datasetsVisible.value = false;
+    itemsProvider.value = historiesProvider;
+}
+
+function onOk() {
+    emit("onSelect", selected.value);
+    emit("onClose");
```

Comment:
While `onCancel` below closes the modal, this `onClose` does not? I also wonder why the names changed from `onOk` to `onSelect` and from `onCancel` to `onClose`.